### PR TITLE
feature(playground): Deploy every commit to subdomain

### PR DESCRIPTION
**Pull request by NicholasLYang**
## Summary

It can be annoying to play around with people's changes to the formatter. By deploying on every commit and storing in a subdomain, we can try out people's changes to the formatter in the playground.

The only issue I could see is that too many commits can clog up the actions pipeline.

### CODE REVIEW

File: .github/workflows/deploy_playground.yml [hunk 1]
```diff
@@ -5,15 +5,13 @@ name: Deploy Playground
 on:
   workflow_dispatch:
   push:
```
[the remark this answers is not in the document]
Comment (ematipico) [reply .github/workflows/deploy_playground.yml:7]: I would add 

```yml
    paths:
      - 'crates/rome_playground'
```

**Review by ematipico — CHANGES_REQUESTED** [review]
There's something important missing in the workflow.

Few questions:
- if there are two commits, does the new deploy override previous one?
- when do we delete old deploys? what's the policy around it? As we're using AWS, it's up to us to clear the bucket from old files

**Comment by leops** [comment]
Would it be possible to use the same deployment process as the website ? In particular I think having auto-generated deploys of the playground for each pull request would be a great tool to help us trying out the changes to the parser or formatter

File: .github/workflows/deploy_playground.yml [hunk 2]
```diff
@@ -23,11 +21,14 @@ jobs:
       - run: npm install --prefix crates/rome_playground
       - uses: jetli/wasm-pack-action@v0.3.0
       - run: wasm-pack build crates/rome_playground --target web --release
-      - run: npm run build --prefix crates/rome_playground
+      - run: npm run build --prefix crates/rome_playground -- --base=/$GITHUB_SHA/
```
[the remark this answers is not in the document]
Comment (NicholasLYang) [reply .github/workflows/deploy_playground.yml:24]: When we build, we assume that the js files will be hosted off of a subdomain that is the commit hash, i.e. play.rome.tools/e2bd79ed0e66acb2049436837ee6e4c6594ec549/index.html

**Comment by ematipico** [comment]
> Would it be possible to use the same deployment process as the website ? In particular I think having auto-generated deploys of the playground for each pull request would be a great tool to help us trying out the changes to the parser or formatter

This is already in place. Every time you open a PR, there's a cloudflare hook that adds a comment to the PR with a URL to the built website: 
Screenshot 2022-04-06 at 15 45 42

**Comment by NicholasLYang** [comment]
@leops Unfortunately Cloudflare Pages doesn't allow multiple sites in the same repository. This is effectively mimicking CF Pages by building on every commit

**Comment by NicholasLYang** [comment]
If we kept the playground as a separate repo, we could just use pages. We could try having everything built at once I suppose, website and playground, but we'd need to embed the playground in the website.

**Comment by ematipico** [comment]
> @leops Unfortunately Cloudflare Pages doesn't allow multiple sites in the same repository. This is effectively mimicking CF Pages by building on every commit

That's because we assume we have a different domain for the playground. What about a `/playground` directory where we copy over the built playground? Once we have that, we just need to update the the npm commands on cloudflare

**Comment by NicholasLYang** [comment]
@ematipico  Just spent some time trying to set this up. I remembered that Cloudflare Pages doesn't support wasm-pack. Before when I was using pages with the playground, I had to manually run wasm-pack on my local machine. This might be our best option for now, until pages adds wasm-pack

File: .github/workflows/deploy_playground.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ematipico) [reply .github/workflows/deploy_playground.yml:24]: Who's the consumer of `base`? I can't understand

**Comment by ematipico** [comment]
Do you plan to print a git comment later? Because as it is now, it's really hard to understand what's the published url

**Comment by NicholasLYang** [comment]
yeah I was thinking of making a bot

**Comment by ematipico** [comment]
> yeah I was thinking of making a bot

There's no need. We already have a bunch of actions and publish and update the comments. We just to copy it and personalise the message. I wanted to know when do you plan to do that: this PR or another one

File: .github/workflows/deploy_playground.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ematipico) [reply .github/workflows/deploy_playground.yml:7]: Sorry, I am not sure I follow. How does my suggestion prevent that?

**Comment by ematipico** [comment]
> @ematipico Just spent some time trying to set this up. I remembered that Cloudflare Pages doesn't support wasm-pack. Before when I was using pages with the playground, I had to manually run wasm-pack on my local machine. This might be our best option for now, until pages adds wasm-pack

Just leaving a suggestion. Another way might be to track in the repository the built wasm file. So we have to build it ourselves and commit the changes.

File: .github/workflows/deploy_playground.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (NicholasLYang) [reply .github/workflows/deploy_playground.yml:27]: `vite`. The flag is telling vite to write all the urls with the offset of the commit hash

**Comment by NicholasLYang** [comment]
@ematipico  That's what I did before. It's not really ideal because it's very easy to forget. We could have a git hook installed but people would need to add it manually to their local setups.

**Review by ematipico — APPROVED** [review]

**Pull request merged**